scripts/runnable_recipe_ray_vllm_weight_sync.py

```diff
@@ -368,6 +369,447 @@ def run(self):
+            def get_logprob(output):
+                t = []
+                for v, tid in zip(output.logprobs, output.token_ids):
+                    t.append(
+                        v[tid]["logprob"] if v[tid].get("logprob") is not None else 1.0
+                    )
+                return torch.tensor(t)
+
+            if output.logprobs:
+                output.logprobs = get_logprob(output)
+            output.token_ids = torch.tensor(output.token_ids)
+            return output
+
+        if isinstance(self.outputs, list):
+            outputs = self.outputs
+            outputs = [
+                postproc(from_dataclass(output, dest_cls=CompletionOutput_tc))
+                for output in outputs
+            ]
+            if len(outputs) == 1:
+                self.outputs = outputs[0]
+            else:
+                self.outputs = maybe_dense_stack(outputs)
+            if self.prompt_logprobs is not None:
+                self.prompt_logprobs = torch.tensor(
+                    [
+                        v[tid].logprob if v is not None else 0.0
+                        for v, tid in _zip_strict(
+                            self.prompt_logprobs, self.prompt_token_ids
+                        )
+                    ]
+                )
+            self.prompt_token_ids = torch.tensor(self.prompt_token_ids)
+            self.num_cached_tokens = torch.tensor(self.num_cached_tokens)
+
+    @classmethod
+    def from_request_output(cls, requests):
+        out = lazy_stack(
+            [
+                cls(
+                    request_id=request.request_id,
+                    prompt=request.prompt,
+                    prompt_token_ids=request.prompt_token_ids,
+                    prompt_logprobs=request.prompt_logprobs,
+                    outputs=request.outputs,
+                    finished=request.finished,
+                    metrics=request.metrics,
+                    lora_request=request.lora_request,
+                    encoder_prompt=request.encoder_prompt,
+                    encoder_prompt_token_ids=request.encoder_prompt_token_ids,
+                    num_cached_tokens=request.num_cached_tokens,
+                )
+                for request in requests
+            ]
+        )
+        return out
+
+def _get_output_tokens_and_log_probs(tokens_out, tokenizer, log_prob_key, token_response_key, text_response_key, padding_value):
+    tokens_out = _RequestOutput_tc.from_request_output(tokens_out)
+
+    tokens_response_td = tokens_out.outputs._tensordict.select(
+        "text", "token_ids", "logprobs", strict=False
+    )
+
+    # ================ added padding ===========================================
+    tokens_response_td = tokens_response_td.densify(
+                layout=torch.strided
+    ).to_padded_tensor(padding=padding_value)
+
+    padded_values = tokens_response_td["token_ids"] == padding_value
+    if padded_values.any():
+        lps = tokens_response_td["logprobs"]
```

Comment:
Sets the logprob of every padded value to 1.0?

Reply:
I believe this currently happens in the base script, is that not true?